src/Context/products-context.js

```diff
@@ -0,0 +1,348 @@
+              })
+                .then((res) => res.json())
+                .then((data) => console.log(data))
+                .catch((err) => console.log(err));
+            }
+          } else {
+            fetch(
+              `/api/user/playlists/${action.payload.playlistId}/${action.payload.video._id}`,
+              {
+                method: "DELETE",
+                headers: {
+                  "Content-Type": "application/json",
+                  Accept: "application/json",
+                  authorization: encodedToken,
+                },
+              }
+            )
+              .then((res) => res.json())
+              .then((data) => console.log(data))
+              .catch((err) => console.log(err));
+            playlist.videos = playlist.videos.filter(
+              (element) => element._id !== action.payload.video._id
+            );
+          }
+        }
+        return true;
+      });
+      return {
+        ...state,
+        playlistsArray: state.playlistsArray,
+      };
+    case "Remove from Playlist":
+      state.playlistsArray.map((playlist) => {
+        if (playlist.id === action.payload.playlistId) {
+          fetch(
+            `/api/user/playlists/${action.payload.playlistId}/${action.payload.videoId}`,
+            {
+              method: "DELETE",
+              headers: {
+                "Content-Type": "application/json",
+                Accept: "application/json",
+                authorization: encodedToken,
+              },
+            }
+          )
+            .then((res) => res.json())
+            .then((data) => console.log(data))
+            .catch((err) => console.log(err));
+          playlist.videos = playlist.videos.filter(
+            (video) => video.id !== action.payload.videoId
+          );
+        }
+        return true;
+      });
+      return {
+        ...state,
+        playlistsArray: state.playlistsArray,
+      };
+    case "Handle Logout":
+      return {
+        likedArray: [],
+        dislikedArray: [],
+        watchLaterArray: [],
+        historyArray: [],
+        playlistsArray: [],
+      };
+    default:
+      return state;
+  }
+};
+
+const ProductsProvider = ({ children }) => {
+  var likedVideosArray = JSON.parse(localStorage.getItem("LIKED_ARRAY"));
+
+  if (likedVideosArray === null) likedVideosArray = [];
+
+  var dislikedVideosArray = JSON.parse(localStorage.getItem("DISLIKED_ARRAY"));
+
+  if (dislikedVideosArray === null) dislikedVideosArray = [];
+
+  var watchLaterVideosArray = JSON.parse(
+    localStorage.getItem("WATCH_LATER_ARRAY")
+  );
+
+  if (watchLaterVideosArray === null) watchLaterVideosArray = [];
+
+  var historyVideosArray = JSON.parse(localStorage.getItem("HISTORY_ARRAY"));
+
+  if (historyVideosArray === null) historyVideosArray = [];
+
+  var playlistsVideosArray = JSON.parse(
+    localStorage.getItem("PLAYLISTS_ARRAY")
+  );
+
+  if (playlistsVideosArray === null) playlistsVideosArray = [];
+
```

Comment:
you can use `const` here since the token won't change, and since it is a side effect do it in useEffect